drivers/hwinfo/hwinfo_andes.c

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2021 Andes Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/hwinfo.h>
+#include <string.h>
+#include <soc.h>
+
+#define ANDES_RESET_STATUS_MASK BIT_MASK(5)
+
+ssize_t z_impl_hwinfo_get_device_id(uint8_t *buffer, size_t length)
+{
+	uint32_t ver = sys_read32(SMU_BASE + SMU_SYSTEMVER) >> 8;
```

Comment:
1. ATC SMU is the abbreviation of andestech(ATC) and system management unit(SMU).
2. SMU is really a syscon area. But in our scenario, the common region in syscon area that cache and hwinfo drivers access are read-only. 
3. We are aware that having syscon driver might be a better solution for this, but we think we'll make this improvement in other PR

Reply:
@carlocaione in my opinion it is ok. It satisfies the API from a user point of perspective. The driver is not instantiated by the device tree.

Reply:
@alexanderwachter  Nope, this makes no sense. The ATC SMU is a syscon area, we have an API for that and having two different drivers bypassing everything and just reading an address from DT and using that as base by two different drivers is just spaghetti code.

@wtlee1995 no, this could not be an improvement simply because when the syscon driver will be in the picture you will have to basically rewrite this driver and the cache driver.

Still a NACK from me.

Reply:
@carlocaione so what is your proposal? Not using a device tree node? Or not using the hwinfo API?

Reply:
@alexanderwachter the ATC SMU should be in the DT and a proper syscon driver should be writter for that (or just use the generic driver already in place). The hwinfo and cache drivers should use the syscon driver to interface with the hardware.

Reply:
Why should we introduce a dependency to just read a register? I'm more a friend on keeping dependencies at a minimum.

Reply:
Because it's not just a register. This is the second full-fledged driver using the same trick (the other one being the cache driver that is already in the tree).

Also the SMU region (AFAICT) is also used for pinmux, power control and other things. So, do we want to keep just ignoring it and maybe adding even more drivers using the same shortcut?

Having a `smu.h` makes no sense, and it was a mistake accepting it in the first place.

Reply:
@carlocaione lets escalate it to the architecture meeting.